src/Context.php

```diff
@@ -2,27 +2,135 @@
 
 namespace Castor;
 
-use ArrayObject;
-
-/**
- * @template TValue
- *
- * @template-extends ArrayObject<(int|string), TValue>
- */
-class Context extends \ArrayObject
+class Context implements \ArrayAccess
 {
-    public string $currentDirectory;
+    public readonly string $currentDirectory;
 
     /**
-     * @param array<(int|string), TValue> $array The input parameter accepts an array or an Object
+     * @param array<(int|string), mixed> $data The input parameter accepts an array or an Object
      * @param array<string, string> $environment a list of environment variables to add to the command
      */
     public function __construct(
-        public array $array = [],
-        public array $environment = [],
+        public readonly array $data = [],
+        public readonly array $environment = [],
+        string $currentDirectory = null,
+        public readonly bool $tty = false,
+        public readonly bool $pty = true,
+        public readonly float|null $timeout = 60,
     ) {
-        parent::__construct($array, \ArrayObject::ARRAY_AS_PROPS);
+        $this->currentDirectory = $currentDirectory ?? PathHelper::getRoot();
```

Comment:
This can be moved to the constructor argument, isn't?

Reply:
expression is not allowed as a default parameter value